src/features/smSystem/suppliersOrders/routers/OrderDetailsPage.tsx

```diff
@@ -0,0 +1,212 @@
+import { Button, CircularProgress, Stack, TextField, Typography } from '@mui/material';
+import { useState } from 'react';
+
+import { useGetOrderDetails } from '../api/useGetOrderDetails';
+import ProductDetailsInOrderTable from '../tables/ProductDetailsInOrderTable';
+import ProductsInOrderTable from '../tables/ProductsInOrderTable';
+
+const generateTxtFile = (content: string, fileName: string) => {
+  const blob = new Blob([content], { type: 'text/plain' });
+  const url = URL.createObjectURL(blob);
+  const link = document.createElement('a');
+  link.href = url;
+  link.download = fileName;
+  link.click();
+  URL.revokeObjectURL(url);
+};
+
+export const OrderDetailsPage = () => {
+  const getIdFromUrl = () => {
+    const url = window.location.hash;
+    const match = url.match(/orders\/(\d+)/);
+
+    if (match) {
+      const id = parseInt(match[1], 10);
+      return Number.isInteger(id) ? id : 0;
+    }
+    return 0;
+  };
```

Comment:
instead of sth like that use prepared tool for it: `useParams`  from `react-router-dom` lib

```
import { useParams } from 'react-router-dom';
...
interface RouteParams {
id: string; // you can't define it as number because reading it from URL it's always string, you can convert it later if needed
}
....
const { id } = useParams<RouteParams>();
```